airbyte-db/db-lib/src/main/java/io/airbyte/db/bigquery/BigQueryDatabase.java

```
@@ -160,20 +159,35 @@ public List<Table> getProjectTables(final String projectId, final BorneoPaginati
     final BigQuery.TableListOption childPageToken = BigQuery.TableListOption.pageToken(pageOptions.childPageToken());
     LOGGER.info("pagination input {} {} {} {}", parentPageSize, parentPageToken, childPageToken, childPageSize);
     List<TableId> tableIds = new ArrayList<>();
-    bigQuery.listDatasets(projectId, parentPageSize, parentPageToken)
-            .getValues()
-            .forEach(dataset -> bigQuery.listTables(dataset.getDatasetId(), childPageSize, childPageToken)
-                    .getValues()
-                    .forEach(table -> tableIds.add(table.getTableId())));
+    List<Dataset> datasets = new ArrayList<>();
+    try {
+      bigQuery.listDatasets(projectId, parentPageSize, parentPageToken).getValues()
+              .forEach((d) -> datasets.add(d));
+    } catch (final Exception e) {
+      LOGGER.error("Error fetching datasets", e);
+    }
```

Ignoring all errors during the discovery call can have unintended consequences. I.e. if the bigQuery.listDatasets call fails temporarily for any reason, e.g. because the query exceeds some quota, then the connector returns an empty table set without any indication that there was an error. During the subsequent data catalog sync that will then clear out all the data sets and tables from our catalog as well.

I think if bigQuery.listDatasets in particular fails, then we should return an error. And if any of the bigQuery.listTables(<datasetId>) or bigQuery.getTable(tableId) calls fail, maybe should indicate in the response which datasets and/or tables had errors so we can skip them when syncing the catalog?

(Not sure if the latter is even possible, since the Airbyte data catalog format might not support returning extensive error information like this.)

---

removed this